### PR TITLE
adds support for type aliases (typedefs in closure).

### LINKED ISSUE #34: add support for type aliases
in closure

```
/**
 * @typedef {A}
 */
goog.namespace.foo;
```

should output 

```
declare namespace internal.goog.namespace {
  type foo = A;
}
...
```

### PR DESCRIPTION
**Pull request by rkirov**
Closes #34

### CODE REVIEW

File: src/main/java/com/google/javascript/cl2dts/DeclarationGenerator.java [hunk 1]
```diff
@@ -20,18 +20,7 @@
...
-import com.google.javascript.rhino.jstype.JSType;
-import com.google.javascript.rhino.jstype.NamedType;
-import com.google.javascript.rhino.jstype.NoType;
-import com.google.javascript.rhino.jstype.ObjectType;
-import com.google.javascript.rhino.jstype.ProxyObjectType;
-import com.google.javascript.rhino.jstype.TemplateType;
-import com.google.javascript.rhino.jstype.TemplatizedType;
-import com.google.javascript.rhino.jstype.UnionType;
-import com.google.javascript.rhino.jstype.Visitor;
+import com.google.javascript.rhino.jstype.*;
```
Comment (alexeagle) [inline src/main/java/com/google/javascript/cl2dts/DeclarationGenerator.java]: you should grab the google style guide
https://docs.google.com/document/d/1GIdeE-sevA8h5SipHoC1r9YPN7MAI0rnm6vMWYyuJZQ/view#heading=h.v8ow5xu1iioa
then IJ will make your imports match the existing code

File: src/main/java/com/google/javascript/cl2dts/DeclarationGenerator.java [hunk 2]
```diff
@@ -371,6 +362,15 @@ private void walk(boolean isDefault) {
...
           return;
         }
+        // Closure keeps type NONE for the symbol which became the type alias.
+        // However, the aliased type is present in the registry under the same name.
+        if (type.isNoType()) {
+          JSType registryType = typeRegistry.getType(symbol.getName());
+          if (registryType != null) {
+            visitTypeAlias(registryType, getUnqualifiedName());
+            return;
+          }
```
Comment (alexeagle) [inline src/main/java/com/google/javascript/cl2dts/DeclarationGenerator.java]: opt: throw if `type.isNoType` but we didn't return already

File: src/test/java/com/google/javascript/cl2dts/type_alias.js [hunk 3]
```diff
@@ -0,0 +1,6 @@
+goog.provide('typedefs');
+
+/**
+ * @typedef {function(string): string}
+ */
+typedefs.strToStr;
```
Comment (alexeagle) [inline src/test/java/com/google/javascript/cl2dts/type_alias.js]: missing trailing newline

**Comment by alexeagle** [comment]
Nice! LGTM

File: src/test/java/com/google/javascript/cl2dts/type_alias.js (hunk 3 above)
Comment (rkirov) [inline src/test/java/com/google/javascript/cl2dts/type_alias.js]: done.

File: src/main/java/com/google/javascript/cl2dts/DeclarationGenerator.java (hunk 2 above)
Comment (rkirov) [inline src/main/java/com/google/javascript/cl2dts/DeclarationGenerator.java]: I am not that convinced NoType only appears with @typedef, so just rewording the comment for now.

File: src/main/java/com/google/javascript/cl2dts/DeclarationGenerator.java (hunk 1 above)
Comment (rkirov) [inline src/main/java/com/google/javascript/cl2dts/DeclarationGenerator.java]: Thanks! I was wondering how to do that.

**Pull request merged**